mlflow/utils/async_logging/async_logging_queue.py

```diff
@@ -88,16 +88,22 @@ def end_async_logging(self) -> None:
             # the remaining data.
             self._status = QueueStatus.TEAR_DOWN
 
-    def flush(self) -> None:
-        """Flush the async logging queue.
-
-        Calling this method will flush the queue to ensure all the data are logged.
+    def shut_down_async_logging(self) -> None:
```

Comment:
I tried something similar but I wasn't able to clean up the zombie threads. Wondering what's missing.